components/push/src/internal/subscriber.rs

```diff
@@ -186,39 +216,44 @@ impl PushManager {
         if channel_id.is_empty() {
             return Ok(false);
         }
-        let conn = self.make_connection()?;
-        if conn.uaid.is_none() {
-            return Err(PushError::GeneralError(
-                "No subscriptions created yet.".into(),
-            ));
-        }
-        conn.unsubscribe(channel_id)?;
+
+        let (uaid, auth) = self.ensure_auth_pair()?;
+        self.connection.unsubscribe(channel_id, uaid, auth)?;
         self.store.delete_record(channel_id)
     }
 
     pub fn unsubscribe_all(&mut self) -> Result<()> {
-        // make connection before deleting, because deletion kills our uaid/auth meta.
-        let mut conn = self.make_connection()?;
         self.store.delete_all_records()?;
-        conn.unsubscribe_all()?;
+        let (uaid, auth) = self.ensure_auth_pair()?;
+
+        self.connection.unsubscribe_all(uaid, auth)?;
         Ok(())
     }
 
     pub fn update(&mut self, new_token: &str) -> error::Result<bool> {
-        self.store.set_registration_id(new_token)?;
-        let mut conn = self.make_connection()?;
-        // It's OK if we don't have a uaid yet - that means we can't have any subscriptions,
-        // and we've saved our registration_id, so will use it on our first subscription.
-        if conn.uaid.is_none() {
+        if self.registration_id.as_deref() == Some(new_token) {
```

Comment:
We shouldn't send a network request if we're already up to date! If the server loses our registration everything collapses anyways and we can recover through the existing recovery mechanism (verify_connection)